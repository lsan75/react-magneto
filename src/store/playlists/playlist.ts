export interface Track {
  id: number;
  band: string;
  title: string;
  picture: string;
  url: string;
}

export interface Playlist {
  id: number;
  title: string;
  tracks: Track[];
}

export interface PlaylistState {
  playlists: Playlist[];
}

export const defaultPlaylistState = {
  playlists: []
};
