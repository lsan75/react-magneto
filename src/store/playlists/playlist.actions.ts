import { Action, Dispatch } from 'redux';
import axios, { AxiosResponse } from 'axios';
import { Playlist } from './playlist';

axios.defaults.baseURL = 'http://localhost:3000';

export enum PLAYLIST {
  PLAYLIST_FETCH_SUCCESS = 'PLAYLIST_FETCH_SUCCESS'
}

export interface PlaylistAction extends Action {
  payload: Playlist[];
}

export function getPlaylists() {
  return (dispatch: Dispatch<PlaylistAction>): void => {
    axios.get('/playlists').then((value: AxiosResponse<Playlist[]>) => {
      dispatch(loadPlaylists(value.data));
    }).catch(reason => {
      window.console.log(reason);
    });
  };
}

export function loadPlaylists(playlists: Playlist[]): PlaylistAction {
  return {
    type: PLAYLIST.PLAYLIST_FETCH_SUCCESS,
    payload: playlists
  };
}