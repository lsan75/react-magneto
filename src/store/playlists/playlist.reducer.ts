import { PlaylistState, defaultPlaylistState } from './playlist';
import { PLAYLIST, PlaylistAction } from './playlist.actions';

export function playlist(state: PlaylistState = defaultPlaylistState, action: PlaylistAction) {

  switch (action.type) {

    case PLAYLIST.PLAYLIST_FETCH_SUCCESS:
      return Object.assign({}, state, { playlists: action.payload });

    default:
      return state;
  }

}
