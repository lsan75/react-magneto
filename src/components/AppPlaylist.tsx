import * as React from 'react';
import { Playlist } from '../store/playlists/playlist';

interface Props {
  playlists: Playlist[];
}

export class AppPlaylist extends React.Component<Props>  {

  constructor(props: Props) {
    super(props);
  }

  render() {
    const { playlists } = this.props;
    const playlist = playlists.map((item: Playlist) => (
      <li key={item.id}>
        {item.title}
      </li>
    ));
    return (
      <ul>{playlist}</ul>
    );
  }
}
