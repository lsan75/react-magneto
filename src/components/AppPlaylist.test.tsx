import * as React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { AppPlaylist } from './AppPlaylist';
import { Playlist } from '../store/playlists/playlist';

describe('Playlist', () => {
  it('should render correctly', () => {

    const playlists: Playlist[] = [{
      id: 1,
      title: 'pan',
      tracks: []
    }];

    const output = shallow(
      <AppPlaylist playlists={playlists} />
    );

    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
