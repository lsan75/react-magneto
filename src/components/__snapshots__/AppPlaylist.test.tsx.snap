// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist should render correctly 1`] = `
<ul>
  <li
    key="1"
  >
    <Link
      replace={false}
      to="/detail/1"
    >
      pan
    </Link>
  </li>
</ul>
`;
